mkt/webapps/tasks.py

```diff
@@ -285,7 +285,8 @@ def _update_manifest(id, check_hash, failed_fetches):
         webapp.set_iarc_storefront_data()
 
 
-@task
+@post_request_task
+@write
 def update_cached_manifests(id, **kw):
```

Comment:
This part is the real fix, everything else is just adjusting mocks because of the way `@post_request_task` spawns sub-tasks.